bindings/go/evmc/evmc.go

```diff
@@ -0,0 +1,258 @@
+// EVMC: Ethereum Client-VM Connector API.
+// Copyright 2018 Pawel Bylica.
+// Licensed under the MIT License. See the LICENSE file.
+
+package evmc
+
+/*
+#cgo CFLAGS:  -I${SRCDIR}/../../../include
+#cgo LDFLAGS: -ldl
+
+#include <evmc/evmc.h>
+#include <evmc/helpers.h>
+#include <evmc/loader.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+inline int set_option(struct evmc_instance* instance, _GoString_ name, _GoString_ value)
+{
+	if (instance->set_option)
+		return instance->set_option(instance, name.p, value.p);
+	return -1;
+}
+
+struct extended_context
+{
+    struct evmc_context context;
+    int64_t index;
+};
+
+extern const struct evmc_context_fn_table fn_table;
+
+static struct evmc_result execute_wrapper(struct evmc_instance* instance,
+                                          int64_t context_index,
+                                          enum evmc_revision rev,
+                                          _GoBytes_ destination,
+                                          _GoBytes_ sender,
+                                          _GoBytes_ value,
+                                          _GoBytes_ input,
+                                          _GoBytes_ code_hash,
+                                          int64_t gas,
+                                          int32_t depth,
+                                          uint32_t flags,
+                                          _GoBytes_ code)
+{
+	struct evmc_message msg;
+	msg.input_data = input.p;
+	msg.input_size = input.n;
+	msg.gas = gas;
+	msg.depth = depth;
+	msg.kind = EVMC_CALL;
+	msg.flags = flags;
+
+	memcpy(msg.destination.bytes, destination.p, sizeof(msg.destination));
+	memcpy(msg.sender.bytes, sender.p, sizeof(msg.sender));
+	memcpy(msg.value.bytes, value.p, sizeof(msg.value));
+	memcpy(msg.code_hash.bytes, code_hash.p, sizeof(msg.code_hash));
```

Comment:
1. I'm not sure using a C assert is good idea here.
2. Because only this Go module can use this function we know exactly what size it is (still prone to future mistakes).
3. Because this does not compile with Go 1.9, this might be rewritten anyway.